raster-foundry/raster-foundry
# Add task status filter to annotation projects

## Overview

This PR adds a `taskStatusesInclude` QP to the annotation project list endpoint `/api/annotation-projects`, so that filtering on one or multiple task statuses is supported. It also adds project task summaries to the return of each annotation project as a result of the call to the annotation project list endpoint (so that GW frontend does not need to make several requests for progress info on the project list page).

For example, if we currently want to find incomplete projects on GW frontend (e.g. projects with unlabeled tasks), we pass `taskStatusesInclude=UNLABELED` to the api as a QP, the endpoint will then return projects with unlabeled tasks. If in the future (after we support validation in GW), we want to find incomplete projects, and assuming "incomplete" means not labeled, or labeled but not validated, we can pass `taskStatusesInclude=UNLABELED&taskStatusesInclude=LABELED` to the api as QPs.

### Checklist

- ~Swagger specification updated~ (We do not have API spec for annotation project related endpoints)
- ~New tables and queries have appropriate indices added~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- [X] Any new SQL strings have tests (updated the existing test)
- ~Any new endpoints have scope validation and are included in the integration test csv~

## Testing Instructions
- **Start of update on Mar. 24th**
- Run migration
- **End of update on Mar. 24th**
- CI should pass since this PR updates tests that cover this QP use case
- Make sure the changes make sense to you
- If you have annotation projects in your DB, assemble api jar, spin up servers
- try `GET` to `/api/annotation-projects?taskStatusesInclude=<status 1>&taskStatusesInclude=<status2>&...` to see if it returns projects that reflect the task statuses

This is to support frontend in https://github.com/raster-foundry/annotate/issues/727
